dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/PatientController.cs

```diff
@@ -109,18 +156,21 @@ private static IList<TimelineEntryViewModel> ConvertToTimelineEntryViewModels(Ti
                         if (schedule.LocalDateTime >= snapshot.EffectiveStartInstant.InUtc().LocalDateTime &&
                             schedule.LocalDateTime <= snapshot.EffectiveEndInstant.InUtc().LocalDateTime)
                         {
-                            // Add a timeline entry for the schedule
+                            // Add a timeline entry for the schedule. 
+                            // Count all occurences for a frequency task, and only in-window one for a scheduled task
                             timelineEntries.Add(new TimelineEntryViewModel
                             {
                                 TaskId = task.TaskId,
                                 TaskName = task.TaskName,
                                 TaskImageUrl = task.TaskImageUrl,
                                 LocalDateTime = schedule.LocalDateTime,
-                                ScheduleType = schedule.Type
+                                ScheduleType = schedule.Type,
+                                CompletionMetrics = snapshot.CompletionMetrics,
+                                OccurrenceCount = schedule.Occurrences?.Count(o => o.InWindow || snapshot.CompletionMetrics.CompletionType == ActionPlanTaskCompletionType.Frequency) ?? 0
                             });
 
-                            // Add any out-of-window task occurrences to show on the timeline
-                            schedule.Occurrences?.Where(o => !o.InWindow).ToList().ForEach(occurrence => AddOccurrence(timelineEntries, occurrence, task)); ;
+                            // Add any out-of-window task occurrences for scheduled tasks to show on the timeline
+                            schedule.Occurrences?.Where(o => !o.InWindow && snapshot.CompletionMetrics.CompletionType == ActionPlanTaskCompletionType.Scheduled).ToList().ForEach(occurrence => AddOccurrence(timelineEntries, occurrence, task)); ;
```

Comment:
I syntactically prefer LINQ which leaves the `ToList()` for the end. To accomplish this the `.ForEach(...)` would be swapped with a `.Select(...)` and the `ToList()` moved to the end. Both effectively accomplish the same thing however in my opinion the _English_ reads more fluidly that way.